lisa/platform_.py

```diff
@@ -94,6 +94,7 @@ def prepare_environments(self, environments: Environments) -> List[Environment]:
             is_success = self._prepare_environment(environment, log)
             if is_success:
                 prepared_environments.append(environment)
+                environment.reset_capability()
```

Comment:
The capability is changed from a `search space` to a `specifed number` after prepared an environment. So it need to be recalculate. After checked current implementation, it looks no need to cache capability. So that remove reset, and cached capability. The logic is simpler now.